modules/backend/widgets/Lists.php

```diff
@@ -540,8 +543,13 @@ protected function getRecords()
             $records = $model->getNested();
         }
         elseif ($this->showPagination) {
-            $method = $this->showPageNumbers ? 'paginate' : 'simplePaginate';
-            $records = $model->{$method}($this->recordsPerPage, $this->currentPageNumber);
+            $method            = $this->showPageNumbers ? 'paginate' : 'simplePaginate';
+            $currentPageNumber = $this->currentPageNumber;
+            // Restore the last visited page from the session if available.
+            if ( ! $currentPageNumber && $currentPageNumber = $this->getSession('lastVisitedPage')) {
+                $this->putSession('lastVisitedPage', false);
```

Comment:
why store an explicit false to the `lastVisitedPage` session var?

Reply:
Yes, good question. This was from my first implementation. I wanted to restore the page number only the first time I return to the list but I just noticed that all the other settings are saved over multiple page views as well so maybe it would be more consistent if we just store the number until the session ends.

What do you think?

```php
            $currentPageNumber = $this->currentPageNumber;
            if ( ! $currentPageNumber) {
                // Restore the last visited page from the session if available.
                $currentPageNumber = $this->getSession('lastVisitedPage');
            }
```